### ISSUE TITLE
[FEATURE] Native JSON support

### ISSUE THREAD

**Issue by dblock** [opensearch-project/opensearch-go#331]
### Is your feature request related to a problem?

Ingesting a document today requires supplying it as an `IoReader`. Starting with an example in https://xeiaso.net/blog/elasticsearch:

```go
data, err := json.Marshal(esEntry)
if err != nil {
	log.Fatalf("failed to marshal entry: %v", err)
}

resp, err := es.Index("site-search-kb", bytes.NewBuffer(data), es.Index.WithDocumentID(entry.ID))
```

Ugh.

### What solution would you like?

Automatically marshal objects to/from JSON. Starting with the canonical example from https://go.dev/blog/json I think I want something that doesn't say "IoStreams" everywhere.

```
m := Message{"Alice", "Hello", 1294706395881547000}
es.Index("site-search-kb", m, es.Index.WithDocumentID(1))
```

### Do you have any additional context?

Came from https://xeiaso.net/blog/elasticsearch

**Comment by dblock** [comment]
@Jakob3xD would love your thoughts!

**Comment by YuyaAbo** [comment]
I went through the code to understand the Issue and found `opensearchutil.NewJSONReader`. Is this sufficient to resolve the Issue? Is it enough to resolve the Issue or is there some functionality missing?

This utility function seems to be used as follows.

https://github.com/opensearch-project/opensearch-go/blob/c04e36a91a64755d6205a64326a65667d51d285d/opensearchutil/json_reader_integration_test.go#L55-L62

**Comment by dblock** [comment]
@YuyaAbo Thank you!

1. Can this be somehow automatic? In this case this would be a feature request.
2. Do we document this anywhere? Either way we should document this in the user guide, then we can close this. Would you like to help?

**Comment by YuyaAbo** [comment]
@dblock Yes, it would certainly be a useful addition if it could be automated.

I will add an introduction to `opensearchutil.NewJSONReader` to the user guide. I will send a PR later.

We will leave this issue as is for the purpose of adding functionality.

**Comment by tannerjones4075** [comment]
Should this issue be closed? #341 seems to have committed a merge request to add the functionality unless I am mistaken.

**Comment by dblock** [comment]
Yes, closing.
